### PR TITLE
Begin ts conversion

### PR DESCRIPTION
**Pull request by nhannah**
Beginning of the TS conversion, docgen now works but we need to modify the .md's to follow the docgen suggestion of being in the same folder as the component or they don't show.

### CODE REVIEW

File: src/components/Attachment/AttachmentActions.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { Text, TouchableOpacity } from 'react-native';
-import styled from '@stream-io/styled-components';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
```
Comment (virdesai) [inline src/components/Attachment/AttachmentActions.js:4]: shouldn't all of these be like this
```suggestion
import styled from 'styled-components/native';
```

File: src/context.js [hunk 2]
```diff
@@ -50,10 +50,10 @@ export function withTranslationContext(OriginalComponent) {
   return ContextAwareComponent;
 }
 
-const getContextAwareComponent = function(context, originalComponent) {
+const getContextAwareComponent = function (context, originalComponent) {
```
Comment (virdesai) [inline src/context.js:53]: any issue with these being arrow functions instead?

File: src/components/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -19,16 +19,17 @@ export const Circle = styled(AnimatedView)`
...
-const Spinner = () => {
+const Spinner: React.FC & { themePath: string } = () => {
   const rotateValue = useRef(new Animated.Value(0));
 
   const loop = Animated.loop(
     Animated.timing(rotateValue.current, {
       duration: 800,
       easing: Easing.linear,
       toValue: 1,
+      useNativeDriver: false,
```
Comment (virdesai) [inline src/components/Spinner/Spinner.tsx:32]: doesn't work with native driver?

File: src/components/MessageList/MessageNotification.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ const Container = styled.TouchableOpacity`
   height: 27px;
   width: 112px;
   z-index: 10;
-  margin-bottom: 0;
+  margin-bottom: 0px;
```
Comment (virdesai) [inline src/components/MessageList/MessageNotification.js:17]: isn't this normally 0 by default? so we wouldn't need to specific a margin-bottom?

File: src/components/MessageList/MessageList.js [hunk 5]
```diff
@@ -42,7 +42,7 @@ const ErrorNotification = styled.View`
   flex-direction: column;
   align-items: center;
   z-index: 10;
-  margin-bottom: 0;
+  margin-bottom: 0px;
```
Comment (virdesai) [inline src/components/MessageList/MessageList.js:45]: isn't this 0 by default?

File: src/components/MessageList/DateSeparator.js [hunk 6]
```diff
@@ -10,31 +10,31 @@ const Container = styled.View`
...
+  height: 0.5px;
   background-color: ${({ theme }) => theme.colors.light};
   ${({ theme }) => theme.messageList.dateSeparator.line.css}
 `;
 
 const DateText = styled.Text`
-  margin-left: 5;
-  margin-right: 5;
+  margin-left: 5px;
+  margin-right: 5px;
```
Comment (virdesai) [inline src/components/MessageList/DateSeparator.js:27]:
```suggestion
  margin-horizontal: 5px;
```

File: src/components/MessageList/DateSeparator.js [hunk 7]
```diff
@@ -10,31 +10,31 @@ const Container = styled.View`
   flex-direction: row;
   justify-content: center;
   align-items: center;
-  margin-top: 20;
-  margin-bottom: 20;
+  margin-top: 20px;
+  margin-bottom: 20px;
```
Comment (virdesai) [inline src/components/MessageList/DateSeparator.js:14]:
```suggestion
  margin-vertical: 20px;
```

File: src/components/Message/MessageSimple/MessageTextContainer.js [hunk 8]
```diff
@@ -9,29 +9,29 @@ const TextContainer = styled.View`
...
       ? theme.message.content.textContainer.borderRadiusS
-      : theme.message.content.textContainer.borderRadiusL};
+      : theme.message.content.textContainer.borderRadiusL}px;
   padding: 5px;
-  padding-left: 8;
-  margin-top: 2;
-  padding-right: 8;
+  padding-left: 8px;
+  margin-top: 2px;
+  padding-right: 8px;
```
Comment (virdesai) [inline src/components/Message/MessageSimple/MessageTextContainer.js:28]: also could change the padding on the line above to be `padding-vertical: 5px;`. also, we're not dealing with styled-component sort keys?
```suggestion
  margin-top: 2px;
  padding-horizontal: 8px;
```

File: src/components/Message/MessageSimple/MessageContent.js [hunk 9]
```diff
@@ -82,8 +82,8 @@ const DeletedContainer = styled.View`
 `;
 
 const DeletedText = styled.Text`
-  font-size: 15;
-  line-height: 20;
+  font-size: 15px;
+  line-height: 20px;
```
Comment (virdesai) [inline src/components/Message/MessageSimple/MessageContent.js:86]: 👀 um... the dreaded line height

File: src/components/Message/MessageSimple/MessageAvatar.js [hunk 10]
```diff
@@ -1,18 +1,18 @@
...
+  margin-right: ${({ alignment }) => (alignment === 'left' ? 8 : 0)}px;
+  margin-left: ${({ alignment }) => (alignment === 'right' ? 8 : 0)}px;
   ${({ theme }) => theme.message.avatarWrapper.container.css}
 `;
 
 const Spacer = styled.View`
-  width: 32;
-  height: 28;
+  width: 32px;
+  height: 28px;
```
Comment (virdesai) [inline src/components/Message/MessageSimple/MessageAvatar.js:15]: no sorting on styled-components?
```suggestion
  height: 28px;
  width: 32px;
```

File: src/components/IconBadge.tsx [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
-import styled from '@stream-io/styled-components';
+import { View } from 'react-native';
 
-const Container = styled.View``;
+import { styled } from '../styles/styledComponents';
```
Comment (virdesai) [inline src/components/IconBadge.tsx:4]: is this not how all of the other `styled-component` imports should be?

File: src/components/Channel/Channel.js [hunk 12]
```diff
@@ -85,7 +88,7 @@ const Channel = (props) => {
 
     return () => {
       client.off('connection.recovered', handleEvent);
-      channel.off?.(handleEvent);
+      channel.off && channel.off(handleEvent);
```
Comment (virdesai) [inline src/components/Channel/Channel.js:91]: wouldn't this break on the line before so this isn't necessary here?
```suggestion
      channel.off(handleEvent);
```

File: src/context.js (hunk 2 above)
Comment (nhannah) [reply src/context.js:53]: Yeah linter autofix on save

File: src/components/Message/MessageSimple/MessageTextContainer.js (hunk 8 above)
Comment (nhannah) [reply src/components/Message/MessageSimple/MessageTextContainer.js:28]: added a sorter to the eslint for TS so we can fix them as we convert.

File: src/components/IconBadge.tsx (hunk 11 above)
Comment (nhannah) [reply src/components/IconBadge.tsx:4]: So the plan is to shift to using the typed version as we move files to TS...any place referencing theme will show red ;)

File: src/components/Spinner/Spinner.tsx (hunk 3 above)
Comment (nhannah) [reply src/components/Spinner/Spinner.tsx:32]: Looks like it does, moving it to true in the next branch of the conversion.

**Review by vishalnarkhede — APPROVED** [review]

File: src/components/Message/MessageSimple/MessageStatus.js [hunk 13]
```diff
@@ -36,8 +36,8 @@ const DeliveredCircle = styled.View`
 `;
 
 const CheckMark = styled.Image`
-  width: 8;
-  height: 6;
+  width: 8px;
+  height: 6px;
```
Comment (virdesai) [inline src/components/Message/MessageSimple/MessageStatus.js:40]:
```suggestion
  height: 6px;
  width: 8px;
```

File: src/components/Message/MessageSimple/MessageStatus.js [hunk 14]
```diff
@@ -1,33 +1,33 @@
...
   ${({ theme }) => theme.message.status.deliveredContainer.css};
 `;
 
 const DeliveredCircle = styled.View`
-  width: 16;
-  height: 16;
-  border-radius: 16;
+  width: 16px;
+  height: 16px;
+  border-radius: 16px;
```
Comment (virdesai) [inline src/components/Message/MessageSimple/MessageStatus.js:30]: forget what you said yesterday about abc-ing the styled-component keys

File: src/components/MessageList/MessageList.js [hunk 15]
```diff
@@ -254,21 +258,23 @@ const MessageList = (props) => {
...
+        // Mask for edit state
+        editing && disableWhileEditing && (
+          <TouchableOpacity
+            collapsable={false}
+            onPress={clearEditingState}
+            style={{
+              backgroundColor: 'black',
+              height: '100%',
+              opacity: 0.4,
+              position: 'absolute',
```
Comment (virdesai) [inline src/components/MessageList/MessageList.js:271]: obviously this was already in there before but position absolute doesn't seem like it'd do anything here with the height and width and zIndex set

File: src/components/MessageList/TypingIndicatorContainer.js [hunk 16]
```diff
@@ -1,10 +1,10 @@
 import React, { useContext } from 'react';
-import styled from '@stream-io/styled-components';
+import styled from 'styled-components/native';
 
 import { ChannelContext, ChatContext } from '../../context';
 
 const Container = styled.View`
-  bottom: 0;
+  bottom: 0px;
```
Comment (virdesai) [inline src/components/MessageList/TypingIndicatorContainer.js:7]: we just gonna wait to remove these unneeded values later?

**Review by virdesai — APPROVED** [review]
non-blocking comments

File: src/components/MessageList/TypingIndicatorContainer.js (hunk 16 above)
Comment (nhannah) [reply src/components/MessageList/TypingIndicatorContainer.js:7]: Yeah I didn't touch the js files again after I got the ts of styled components happy, we need to just hit them well as we convert

**Pull request merged**